src/charm.py

```diff
@@ -330,24 +340,10 @@ def _render_conf_file(self) -> str:
         )
         return rendered
 
-    def _push_file(
-        self, dst: Path, src: Optional[Path] = None, content: Optional[str] = None
-    ) -> None:
-        if not content:
-            if not src:
-                raise ValueError("`src` or `content` must be provided.")
-            with open(src) as f:
-                content = f.read()
-        self._container.push(dst, content, make_dirs=True)
-
-    def _push_default_files(self) -> None:
-        for schema_file in self._mappers_local_dir_path.iterdir():
-            self._push_file(self._mappers_dir_path / schema_file.name, src=schema_file)
-
-        for schema_file in self._identity_schemas_local_dir_path.iterdir():
-            self._push_file(
-                self._identity_schemas_default_dir_path / schema_file.name, src=schema_file
-            )
+    @backoff.on_exception(backoff.expo, ApiError, max_tries=5, logger=logger)
```

Comment:
Have two questions about other CRUD operations on ConfigMaps:

- do we need to provide the same backoff retry logic to the other operations, like create, delete, etc.?
- what I can see is that we probably should put the retry logic into the `ConfigMapHandler`. We don't need to render the config file several times due to the k8s client API errors.

Reply:
The main reason for the backoff is that the API call may fail because someone else updated the configMap while we were creating the config. In that case, we need to re-generate the config to take into account anything that was changed.